lessons/cryptography/Encoding/encoding.md

```diff
@@ -0,0 +1,118 @@
+
+![Encoding ASCII with CyberChef (https://gchq.github.io/CyberChef/#recipe=Encode_text('US-ASCII%20(7-bit)%20(20127)')To_Binary('Space',7)&input=SGk)](.rsrc/cyberchef-ascii-to-binary.png)
+
+We can also use CyberChef to decode the binary message **"1001000 1101001"** into an ASCII message, which reveals the original message **"Hi"**. 
+
+![Decoding ASCII with CyberChef (https://gchq.github.io/CyberChef/#recipe=From_Binary('Space',7)Decode_text('US-ASCII%20(7-bit)%20(20127)')&input=MTAwMTAwMCAxMTAxMDAx)](.rsrc/cyberchef-binary-to-ascii.png)
+
+
+## Unicode
+Unicode is a standard for encoding which defines 144,697 characters, including those of modern and historical languages, emojis, and formatting codes. There are three main encoding forms in Unicode called UTF-8, UTF-16, and UTF-32. The numbers represent the number of bits needed per encoded character, meaning that it takes less memory to store UTF-8 encoded characters than UTF-32 encoded characters. UTF-8 is commonly used for HTML, and shares similarities with 8-bit ASCII. The following image shows how the top 50 emojis are encoded. 
+
+![Unicode Top 50 Emojis (https://unicode-table.com/en/sets/top-emoji/)](.rsrc/unicode-emojis.png)
+
+The website [Unicode Character Table](https://unicode-table.com/en/) allows you to look up common sets of unicode encodings or search for individual characters. 
+
+For example, one of the top 50 emojis is the **Face with Tears of Joy** emoji. Looking at https://unicode-table.com/en/1F602/ shows a lot of additional information about encoding of this emoji. In the table below, it can be seen that the UTF-8 encoding of this emoji is **11110000 10011111 10011000 10000010**. 
+
+![Unicode of Face with Tears of Joy Emoji (https://unicode-table.com/en/1F602/)](.rsrc/emoji-encoding-table.png)
+
+Using CyberChef, we can then decode this back into the original emoji. 
+
+![Decoding UTF-8 Emoji with CyberChef (https://gchq.github.io/CyberChef/#recipe=From_Binary('Space',8)Encode_text('UTF-8%20(65001)')&input=MTExMTAwMDAgMTAwMTExMTEgMTAwMTEwMDAgMTAwMDAwMTA)](.rsrc/cyberchef-emoji.png)
+
+
+## Base64
+Base64 is an encoding scheme that is commonly used to represent binary data in ASCII text. 64 unique ASCII characters were selected to represent binary data. Characters that are printable and common in most encoding systems were chosen because it is less likely that these characters would be modified while these characters are being sent and received. The following table shows the 64 chosen characters. 
+
+![Base64 Character Set (https://en.wikipedia.org/wiki/Base64)](.rsrc/base64-table.png) 
+
+Binary data is interpreted in groups of three 8-bit bytes, and each group is then converted into four 6-bit base64 characters. If the last group of binary data is too small, it is padded with 0's. Then, **=** is added as padding to complete the four 6-bit character set.
+
+### Base64 Encoding Example
+As an example, we will encode the following binary representation:  
```

Comment:
"Let's say we have some random binary data from an application that needs to be sent"